packages/SwingSet/test/virtualObjects/test-virtualObjectManager.js

```diff
@@ -297,6 +314,211 @@ test('virtual object operations', t => {
   ]);
 });
 
+test('virtual object gc', t => {
+  const log = [];
+  const {
+    makeKind,
+    dumpStore,
+    setExported,
+    deleteEntry,
+    possibleVirtualObjectDeath,
+  } = makeFakeVirtualObjectManager({ cacheSize: 3, log });
+
+  const thingMaker = makeKind(makeThingInstance);
+  const refMaker = makeKind(makeRefInstance);
+
+  // make a bunch of things which we'll use
+  // all virtual objects are born locally ref'd
+  const things = [];
+  for (let i = 1; i <= 9; i += 1) {
+    things.push(thingMaker(`thing #${i}`));
+  }
+  t.is(log.shift(), `set vom.o+1/1 ${minThing('thing #1')}`);
+  t.is(log.shift(), `set vom.o+1/2 ${minThing('thing #2')}`);
+  t.is(log.shift(), `set vom.o+1/3 ${minThing('thing #3')}`);
+  t.is(log.shift(), `set vom.o+1/4 ${minThing('thing #4')}`);
+  t.is(log.shift(), `set vom.o+1/5 ${minThing('thing #5')}`);
+  t.deepEqual(log, []);
```

Comment:
why are there only 5 set calls when there are 9 things being created?

Reply:
ok, I guess you're fighting the cache

Reply:
The cache is mysterious to those uninitiated in its mysteries.